src/core/pointcloud/qgspointcloudlayer.cpp

```diff
@@ -484,32 +484,14 @@ QString QgsPointCloudLayer::encodedSource( const QString &source, const QgsReadW
 {
   QGIS_PROTECT_QOBJECT_THREAD_ACCESS
 
-  QVariantMap parts = QgsProviderRegistry::instance()->decodeUri( providerType(), source );
-  if ( parts.contains( QStringLiteral( "path" ) ) )
-  {
-    parts.insert( QStringLiteral( "path" ), context.pathResolver().writePath( parts.value( QStringLiteral( "path" ) ).toString() ) );
-    return QgsProviderRegistry::instance()->encodeUri( providerType(), parts );
-  }
-  else
-  {
-    return source;
-  }
+  return QgsProviderRegistry::instance()->absoluteToRelativeUri( mProviderKey, source, context );
```

Comment:
Could these implementations be moved to QgsMapLayer? They are shared by point cloud/ mesh/raster already.

Reply:
I thought about doing that, but in the end decided to skip that for now:
- base map layer class does not do any absolute/relative path conversion - changing that could in theory cause issue(s) in custom plugin layers (?)
- vector layer has an extra odd bit of code for memory layers
- vector tile layer needs forcing provider key to be `vectortile` as earlier versions of QGIS did saved empty provider key

Reply:
I'm fine with that!